src/validation.cpp

```diff
@@ -436,7 +436,7 @@ bool CheckSequenceLocks(const CTxMemPool& pool, const CTransaction& tx, int flag
 // Returns the script flags which should be checked for a given block
 static unsigned int GetBlockScriptFlags(const CBlockIndex* pindex, const Consensus::Params& chainparams);
 
-static void LimitMempoolSize(CTxMemPool& pool, size_t limit, unsigned long age) {
+static void LimitMempoolSize(CTxMemPool& pool, size_t limit, unsigned long age) EXCLUSIVE_LOCKS_REQUIRED(pool.cs) {
```

Comment:
We require the `pool.cs` lock for the duration of this function (instead of just relying on the lock in `pool.Expire`) because we want to keep the mempool from changing while `pcoinsTip` modifications happen, correct?

Reply:
Yeah, calling `LimitMempoolSize` only makes sense after you have written to the mempool, in which case you already have acquired the lock and can keep it for `LimitMempoolSize`.